types/parse5/index.d.ts

```diff
@@ -277,62 +273,95 @@ export interface DefaultTreeTextNode extends DefaultTreeChildNode {
      * Text node source code location info. Available if location info is enabled via {@link ParserOptions}.
      */
     sourceCodeLocation?: Location;
-}
 
-// Generic node interfaces
-/**
- * Generic Node interface.
- * Cast to the actual AST interface (e.g. {@link parse5.DefaultTreeNode}) to get access to the properties.
- */
-export type Node = DefaultTreeNode | object;
-
-/**
- * Generic ChildNode interface.
- * Cast to the actual AST interface (e.g. {@link parse5.DefaultTreeChildNode}) to get access to the properties.
- */
-export type ChildNode = DefaultTreeChildNode | object;
-
-/**
- * Generic ParentNode interface.
- * Cast to the actual AST interface (e.g. {@link parse5.DefaultTreeParentNode}) to get access to the properties.
- */
-export type ParentNode = DefaultTreeParentNode | object;
-
-/**
- * Generic DocumentType interface.
- * Cast to the actual AST interface (e.g. {@link parse5.DefaultTreeDocumentType}) to get access to the properties.
- */
-export type DocumentType = DefaultTreeDocumentType | object;
-
-/**
- * Generic Document interface.
- * Cast to the actual AST interface (e.g. {@link parse5.DefaultTreeDocument}) to get access to the properties.
- */
-export type Document = DefaultTreeDocument | object;
+    /**
+     * Parent node.
+     */
+    parentNode: ParentNode;
+}
 
 /**
- * Generic DocumentFragment interface.
- * Cast to the actual AST interface (e.g. {@link parse5.DefaultTreeDocumentFragment}) to get access to the properties.
+ * Default tree adapter Node interface.
  */
-export type DocumentFragment = DefaultTreeDocumentFragment | object;
+export type Node = CommentNode | Document | DocumentFragment | DocumentType | Element | TextNode;
 
 /**
- * Generic Element interface.
- * Cast to the actual AST interface (e.g. {@link parse5.DefaultTreeElement}) to get access to the properties.
+ * Default tree adapter ChildNode type.
  */
-export type Element = DefaultTreeElement | object;
+export type ChildNode = TextNode | Element | CommentNode;
 
 /**
- * Generic TextNode interface.
- * Cast to the actual AST interface (e.g. {@link parse5.DefaultTreeTextNode}) to get access to the properties.
+ * Default tree adapter ParentNode type.
  */
-export type TextNode = DefaultTreeTextNode | object;
+export type ParentNode = Document | DocumentFragment | Element;
+
+export interface TreeAdapterTypeMap {
+    attribute: any;
```

Comment:
Should these be `unknown`?

Reply:
i suppose this didn't exist before so its impossible anyone would be depending on the weak typing. so may be able to get away with `unknown`, ill give it a go